lib/internal/http2/core.js

```diff
@@ -2038,21 +2043,23 @@ const setTimeout = {
   enumerable: true,
   writable: true,
   value: function(msecs, callback) {
-    if (typeof msecs !== 'number') {
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                 'msecs',
-                                 'number');
-    }
+    // Type checking identical to timers.enroll()
+    msecs = validateTimerDuration(msecs);
+
     if (msecs === 0) {
-      unenroll(this);
+      clearTimeout(this[kTimeout]);
+
       if (callback !== undefined) {
```

Comment:
I feel like these conditionals are probably not how the API should work, but I'll leave that for another PR.